### ISSUE TITLE
teamviewer: GUI doesn't detect the running daemon

### ISSUE THREAD

**Issue by steveeJ** [NixOS/nixpkgs#31421]
## Issue description
The `teamviewer` application doesn't detect the daemon running. Instead the GUI says

> The TeamViewer daemon is not running!
> ...

The daemon *is* actually running, as confirmed by this:

```
steveej-laptop% sudo teamviewer --daemon start

systemctl start teamviewerd.service

steveej-laptop% sudo teamviewer --daemon status

systemctl status teamviewerd.service
● teamviewerd.service - TeamViewer remote control daemon
   Loaded: loaded (/nix/store/zvlqjg708fkxmf5q2ybnaps1kc56mz3j-unit-teamviewerd.service/teamviewerd.service; enabled; vendor preset: enabled)
   Active: active (running) since Thu 2017-11-09 08:09:55 CET; 16min ago
  Process: 7129 ExecStart=/nix/store/mzbzzshgqg7ysiisv4j1v2xmxh4v5xwn-teamviewer-12.0.76279/bin/teamviewerd -d (code=exited, status=0/SUCCESS
  Process: 7127 ExecStartPre=/nix/store/w42z5k8sc9x63d08ynwwpj32nwn3kwxz-unit-script/bin/teamviewerd-pre-start (code=exited, status=0/SUCCESS
 Main PID: 7131 (teamviewerd)
    Tasks: 16 (limit: 4915)
   Memory: 3.0M
      CPU: 705ms
   CGroup: /system.slice/teamviewerd.service
           └─7131 /nix/store/mzbzzshgqg7ysiisv4j1v2xmxh4v5xwn-teamviewer-12.0.76279/bin/teamviewerd -d

Nov 09 08:09:55 steveej-laptop systemd[1]: Starting TeamViewer remote control daemon...
Nov 09 08:09:55 steveej-laptop systemd[1]: teamviewerd.service: PID file /run/teamviewerd.pid not readable (yet?) after start: No such file o
Nov 09 08:09:55 steveej-laptop systemd[1]: Started TeamViewer remote control daemon.
```

The PID file does indeed exist:

```
steveej-laptop% ls -lha /run/teamviewerd.pid
-rw-r--r-- 1 root root 4 Nov  9 08:09 /run/teamviewerd.pid
```

### Steps to reproduce

1. Enable teamviewerd service in the nixos configuration
2. Rebuild and switch/reboot to the new configuration
3. Log in as user and run `teamviewer`

## Technical details
* System: 17.09.1997.1a8a95e879
* Nix version: (run `nix-env --version`)
* Nixpkgs version: "17.09.1997.1a8a95e879"
* Sandboxing enabled: false

**Comment by globin** [comment]
cc @fadenb

**Comment by fadenb** [comment]
@steveeJ can you try with version 12.0.90041 of TeamViewer ( https://dl.tvcdn.de/download/version_12x/teamviewer_12.0.90041_i386.deb ) ? If this version fixes the issue I will package it for backport in 17.09. If not I suggest to drop version 12 in favour of the newly released 13 branch.

**Comment by steveeJ** [comment]
@fadenb Sure, I'll try! Is it already in the unstable channel or in nixpkgs/master?

**Comment by steveeJ** [comment]
I just tried teamviewer-12.0.85001 which has the same problem.

**Comment by yuyomagico** [comment]
Is there any solution for this yet?

**Comment by wamserma** [comment]
@steveeJ Is this still an issue?

**Comment by steveeJ** [comment]
@wamserma 
It works now! Seems like the PR(s) which fixed this didn't have a reference.

**Comment by rolfschr** [comment]
I've had this issue, too. I have (for whatever reason) installed `teamviewer` using `nix-env` and *at the same time* had also set up `services.teamviewer.enable = true;`. This worked fine until one of the two got updated. Then, I got the GUI (`nix-env`) to tell me that the daemon (`services.teamviewer`) doesn't run although it did. The solution here was to simply remove `teamviewer` from the `nix-env`. Then, both versions are in sync again (because `services.teamviewer` installs `teamviewer` anyways).
